tests/cpu/test_kalman_fitter.cpp

```diff
@@ -0,0 +1,193 @@
+    const auto mat = detray::silicon_tml<scalar>();
+    const scalar thickness = 0.5 * detray::unit<scalar>::mm;
+
+    const detector_type det = create_telescope_detector(
+        host_mr,
+        b_field_t(b_field_t::backend_t::configuration_t{B[0], B[1], B[2]}),
+        plane_positions, traj, 100000. * detray::unit<scalar>::mm,
+        100000. * detray::unit<scalar>::mm, mat, thickness);
+
+    /***************************
+     * Prepare track candidates
+     ***************************/
+
+    // Navigator, stepper, and fitter
+    using navigator_type = detray::navigator<decltype(det)>;
+    using rk_stepper_type = detray::rk_stepper<b_field_t::view_t, transform3,
+                                               detray::constrained_step<>>;
+
+    using fitter_type = kalman_fitter<rk_stepper_type, navigator_type>;
+
+    // Standard deviations for seed track parameters
+    std::array<scalar, e_bound_size> stddevs = {
+        0.03 * detray::unit<scalar>::mm,
+        0.03 * detray::unit<scalar>::mm,
+        0.017,
+        0.017,
+        0.001 / detray::unit<scalar>::GeV,
+        1 * detray::unit<scalar>::ns};
+
+    // Seed generator
+    seed_generator<rk_stepper_type, navigator_type> sg(det, stddevs);
+
+    // Fitting algorithm object
+    fitting_algorithm<fitter_type> fitting;
+
+    // File path
+    std::string file_path =
+        std::to_string(p0) + "_GeV_" + std::to_string(phi0) + "_phi/";
+    std::string full_path =
+        "detray_simulation/telescope/kf_validation/" + file_path;
+
+    std::size_t n_events = 100;
+
+    // Iterate over events
+    for (std::size_t i_evt = 0; i_evt < n_events; i_evt++) {
+        // Event map
+        traccc::event_map2 evt_map(i_evt, full_path, full_path, full_path);
+
+        // Truth Track Candidates
+        traccc::track_candidate_container_types::host track_candidates =
+            evt_map.generate_truth_candidates(sg, host_mr);
+
+        // n_trakcs = 100
+        ASSERT_EQ(track_candidates.size(), 100);
+
+        // Run fitting
+        auto track_states = fitting(det, track_candidates);
+
+        // Iterator over tracks
+        const std::size_t n_tracks = track_states.size();
+
+        // n_trakcs = 100
+        ASSERT_EQ(n_tracks, 100);
+        for (std::size_t i_trk = 0; i_trk < n_tracks; i_trk++) {
+
+            const auto& track_states_per_track = track_states[i_trk].items;
+            ASSERT_EQ(track_states_per_track.size(),
+                      plane_positions.size() - 2);
+
+            fit_performance_writer.write(track_states_per_track, det, evt_map);
+        }
+    }
+
+    fit_performance_writer.finalize();
+
+    /********************
+     * Pull value test
+     ********************/
+
+    std::array<std::string, 5> pull_names{"pull_d0", "pull_z0", "pull_phi",
+                                          "pull_theta", "pull_qop"};
+
+    auto f = fit_performance_writer.get_file();
+
+    auto gaus = new TF1("gaus", "gaus", -5, 5);
```

Comment:
You shouldn't create this object on the heap. It just goes and leaks away like this.

You could just declare it on the stack with:

```c++
TF1 gaus{"gaus", "gaus", -5, 5};
```

Or you could even make it a member variable of `KalmanFittingTests`...

Reply:
That was dumb :p Fixed.